### PR TITLE
[Typescript-Node] Mark deprecated endpoints

### PR DESCRIPTION
**Pull request by gcatanese**
Add @deprecated tag when processing deprecated operations in the OpenAPI file

@wing328 I have added a helper method (`assertFileContains(Path path, String line, int lineNumber)`) to verify a specific line (more effective than searching for the string in the file anywhere), let me know if this is ok or you see some drawbacks. Thanks.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04) @joscha (2024/10)

### CODE REVIEW

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/TestUtils.java [hunk 1]
```diff
@@ -173,6 +173,18 @@ public static void assertFileContains(Path path, String... lines) {
         }
     }
 
+    public static void assertFileContains(Path path, String line, int lineNumber) {
+        try {
+            List<String> lines = Files.readAllLines(path);
+            assertNotNull(lines);
+
+            String target = lines.get(lineNumber - 1);
+            assertTrue(target.contains(line), "File does not contain line [" + line + "] at lineNumber [" + lineNumber + "]");
```
Comment (joscha) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/TestUtils.java:182]: If we make this a:

```suggestion
            assertTrue(target.contains(line), "Expected "+ path + ":" + lineNumber + " to contain [" + line + "]");
```

Then links become clickable in the command line

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java [hunk 2]
```diff
@@ -296,6 +296,26 @@ public void testDeprecatedAttribute() throws IOException {
...
+                .setInputSpec("src/test/resources/3_0/typescript-node/SampleProject.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        final ClientOptInput clientOptInput = configurator.toClientOptInput();
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(clientOptInput).generate();
+        //files.forEach(File::deleteOnExit);
+
+        TestUtils.assertFileContains(Paths.get(output + "/api/defaultApi.ts"),
+                "* @deprecated", 118);
```
Comment (joscha) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java:316]: asserting on the line number feels very fragile to me. Any refactor in the mustache that changes any lines before the deprecated tag will require future contributors to update this assertion.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java [hunk 3]
```diff
@@ -296,6 +296,26 @@ public void testDeprecatedAttribute() throws IOException {
...
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("typescript-node")
+                .setInputSpec("src/test/resources/3_0/typescript-node/SampleProject.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        final ClientOptInput clientOptInput = configurator.toClientOptInput();
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(clientOptInput).generate();
+        //files.forEach(File::deleteOnExit);
```
Comment (joscha) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java:313]:
```suggestion
```
?
or put it in a `finally` block?

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache [hunk 4]
```diff
@@ -153,6 +153,10 @@ export class {{classname}} {
      {{#allParams}}
      * @param {{paramName}} {{description}}
      {{/allParams}}
+     {{#isDeprecated}}
+     *
+	 * @deprecated
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:158]:
```suggestion
 * @deprecated
```

File: samples/client/petstore/typescript-node/3_0/api/defaultApi.ts [hunk 5]
```diff
@@ -114,6 +114,8 @@ export class DefaultApi {
      * @param strCode Code as header
      * @param strCode2 Code as header2
      * @param patchUsersUserIdRequest Patch user properties to update.
+     *
+	 * @deprecated
```
Comment (joscha) [inline samples/client/petstore/typescript-node/3_0/api/defaultApi.ts:118]: see above

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java (hunk 3 above)
Comment (gcatanese) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java]: I have missed it. It is not fixed.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java (hunk 2 above)
Comment (gcatanese) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java]: Agree. Now that I think about it, it isn't a great idea. I have removed it.

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache [hunk 6]
```diff
@@ -155,7 +155,7 @@ export class {{classname}} {
      {{/allParams}}
      {{#isDeprecated}}
      *
-	 * @deprecated
+ * @deprecated
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:158]: I think you had tabs. If you use spaces to indent it, it should work.

File: modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache [hunk 7]
```diff
@@ -153,6 +153,10 @@ export class {{classname}} {
      {{#allParams}}
      * @param {{paramName}} {{description}}
      {{/allParams}}
+     {{#isDeprecated}}
+     *
+ * @deprecated
```
Comment (joscha) [inline modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache:158]:
```suggestion
    * @deprecated
```
This should work, it's tricky in the webinterface :)

**Review by joscha — APPROVED** [review]
🎉

**Comment by macjohnny** [comment]
@gcatanese can you please re-generate the samples? then we can merge

**Comment by joscha** [comment]
> @gcatanese can you please re-generate the samples? then we can merge

diff seems unrelated to this PR:

Screenshot 2024-10-03 at 3 01 02 PM

**Comment by joscha** [comment]
5f345c0213cde1f905d0cc012d461e4cd693ec62 was just merged, which fixes the samples

**Comment by macjohnny** [comment]
@gcatanese can you rebase on the latest master branch?

**Comment by joscha** [comment]
[38c5618](https://github.com/OpenAPITools/openapi-generator/pull/19770/commits/38c5618df9d50098945c67b58fac7eb5b65fa522) doesn't look quite right. You have to either fast forward or rebase your changes on top, so you don't get the diff between this branch and the merge target (master). If you revert the last commit and force push, then github should actually show an `Update branch from master` button that should do the right thing.

**Pull request merged**